pkg/network/tracer/tracer_test.go

```diff
@@ -1195,101 +1195,6 @@ func TestConnectedUDPSendIPv6(t *testing.T) {
 	assert.Equal(t, bytesSent, int(outgoing[0].MonotonicSentBytes))
 }
 
-func TestConnectionClobber(t *testing.T) {
```

Comment:
Why did you remove this test? It should be easy to adapt it.

Reply:
sorry, I was actually going to ping you about this to validate whether we still need this or not. Could you elaborate on what we're testing here?

Reply:
This is making sure that a 2nd call to `GetActiveConnections` doesn't destroy/clobber the data returned from the 1st call. The careful choice of the # of connections is to prevent an underlying buffer resize (which definitely wouldn't clobber anything).

Reply:
See https://github.com/DataDog/datadog-agent/pull/6557

Reply:
I think we could simplify this test and just ensure that the result from `GetConnections` don't point to the same same memory address of `buffer.Connections()`. Would that be ok?

Reply:
That may work, but feels a bit implementation specific to me. 

This test was simply ensuring that the results from two consecutive calls didn't interfere with another. If I could've written it without the sizing logic, I would've, since it makes some assumptions about how the underlying buffer works.

Reply:
yeah, that's why I'm asking. I feel that there is already a sort of implementation leak in the test, and I find it a little bit hard to follow, which is why I'd rather rewrite it in a simpler way.

Reply:
> You could copy the return value of buffer.Connections() into a reused buffer and return that buffer each time. Your test would pass, but it would be clobbering results. 
I think I'm not fully understanding the intent of your test then 😅 
Can you elaborate on what "clobbering results" means in this context?

At the very least it sounds to me that this has only to do with `state.GetDelta` so I could at least remove the cruft of creating a TCP server etc

Reply:
clobbering = the value of the return value from `GetActiveConnections` call #1 changing after `GetActiveConnections` call #2. We are doing lots of tricks to save on allocations, but we need to ensure we aren't modifying values that are in-use elsewhere.

The crux of the test is the assertions:
https://github.com/DataDog/datadog-agent/blob/1e940bafc53b855251b8f19f9ad5d0ed39804bcd/pkg/network/tracer/tracer_test.go#L1288-L1289

The fix was indeed in `state.GetDelta`, but I'm treating the implementation like a blackbox as much as possible.

Reply:
```
func TestConnectionClobber(t *testing.T) {
	state := newDefaultState()
	active := []ConnectionStats{
		ConnectionStats{
			Source: util.AddressFromString("127.0.0.1"),
			Dest:   util.AddressFromString("8.8.8.8"),
			SPort:  123,
			DPort:  53,
		},
	}

	delta := state.GetDelta("1", 1, active, nil, nil, nil)
	require.Len(t, delta.Connections, 1)

	c := &delta.Connections[0]
	require.Equal(t, 123, int(c.SPort))

	// Modify result value and verify it doesn't interfere to with the original slice
	c.SPort = 456
	assert.Equal(t, 123, int(active[0].SPort))
}
```
Does this look reasonable?

Reply:
So that tests that `active` and `delta` are decoupled, but doesn't test if two successive calls are coupled in some way. (which I think is the largest risk when trying to save allocations. Actually reusing the same memory as what you returned.)